### PR TITLE
Update to latest flutter_cache_manager and firebase_performance

### PR DESCRIPTION
**Pull request by PGScully**
This PR updates to use the latest version of `flutter_cache_manager` and `firebase_performance`.  It also updates for null safety.

`flutter_cache_manager` has changed significantly, and no longer uses a fileService.

Further work is needed to make sure http response codes, and download sizes are correctly recorder in all cases.

### CODE REVIEW

File: lib/src/traced_cache.dart [hunk 1]
```diff
@@ -1,30 +1,99 @@
...
+        withProgress = false;
+      }
+    } catch (e) {
+      print('TracedCacheManager: Failed to load cached file for $url with error:\n$e');
+    }
+    if (cacheFile == null || cacheFile.validTill.isBefore(DateTime.now())) {
+      final Request req = Request('GET', Uri.parse(url));
+      req.headers.addAll(headers ?? {});
+      final HttpMetric metric =
+          FirebasePerformance.instance.newHttpMetric(req.url.toString(), HttpMethod.Get);
```
Comment (nareddyt) [inline lib/src/traced_cache.dart:52]: nit: No need to toString() the URL. Function should accept URL object

File: lib/src/traced_cache.dart [hunk 2]
```diff
@@ -1,30 +1,99 @@
...
+            in super.webHelper.downloadFile(url, key: key, authHeaders: headers)) {
+          if (response is DownloadProgress && withProgress) {
+            metric.responsePayloadSize = response.totalSize;
+            metric.httpResponseCode = 200;
+            streamController.add(response);
+          }
+          if (response is FileInfo) {
+            streamController.add(response);
+            metric.httpResponseCode = 200;
+            metric.requestPayloadSize = response.file.lengthSync();
```
Comment (nareddyt) [inline lib/src/traced_cache.dart]: this seems incorrect, request size is response length?

File: lib/src/traced_cache.dart [hunk 3]
```diff
@@ -1,30 +1,99 @@
...
+          }
+        }
+      } catch (e) {
+        assert(() {
+          print('TracedCacheManager: Failed to download file from $url with error:\n$e');
+          return true;
```
Comment (nareddyt) [inline lib/src/traced_cache.dart:71]: Hmm, this function should return a Future. Why return `true`?

Comment (nareddyt) [reply lib/src/traced_cache.dart:71]: Im not sure about the assert. If file download fails, will you crash the app? We should handle it gracefully

File: lib/src/traced_cache.dart [hunk 4]
```diff
@@ -1,30 +1,99 @@
...
+    }
+    unawaited(streamController.close());
+  }
+
+  Future<FileInfo> downloadFile(String url,
+      {String? key, Map<String, String>? authHeaders, bool force = false}) async {
+    final Request req = Request('GET', Uri.parse(url));
+    req.headers.addAll(authHeaders ?? {});
+    final HttpMetric metric =
+        FirebasePerformance.instance.newHttpMetric(req.url.toString(), HttpMethod.Get);
```
Comment (nareddyt) [inline lib/src/traced_cache.dart:88]: ditto

File: lib/src/traced_cache.dart [hunk 5]
```diff
@@ -1,30 +1,99 @@
...
+          streamController.addError(e);
+        }
+      } finally {
+        metric.stop();
+      }
+    }
+    unawaited(streamController.close());
+  }
+
+  Future<FileInfo> downloadFile(String url,
```
Comment (nareddyt) [inline lib/src/traced_cache.dart:83]: Can you help explain the difference between this function and `getFileStream`

**Review by nareddyt — CHANGES_REQUESTED** [review]
Thanks for the update! Once the changes are addressed, do I need to publish a new version to pub.dev?

**Comment by PGScully** [comment]
The contents of this file are copied out of `flutter_cache_manager`.  The only changes are the adding of `firebase_performance` metrics.  All the odd nitpicks are from `flutter_cache_manager`.

File: lib/src/traced_cache.dart (hunk 1 above)
Comment (PGScully) [reply lib/src/traced_cache.dart:52]: No, it is expecting a `String`.

File: lib/src/traced_cache.dart (hunk 2 above)
Comment (PGScully) [reply lib/src/traced_cache.dart]: correct, fix pushed

**Review by nareddyt — APPROVED** [review]
Thanks, didn't realize it was the same code

**Comment by nareddyt** [comment]
Do I need to publish a new version to pub.dev?

**Comment by PGScully** [comment]
> Do I need to publish a new version to pub.dev?

If you're happy with it, yes please.

**Pull request merged**